HW3/P3/sum.cl

```diff
@@ -24,20 +30,46 @@ __kernel void sum_coalesced(__global float* x,
     // You can assume get_local_size(0) is a power of 2.
     //
     // See http://www.nehalemlabs.net/prototype/blog/2014/06/16/parallel-programming-with-opencl-and-python-parallel-reduce/
-    for (;;) { // YOUR CODE HERE
-        ; // YOUR CODE HERE
+    
+
+    // tried implementation below, it didn't work
+    // because local_size >> 2 is not the same as log_2(local_size)
```

Comment:
local_size >> 2 should give the same answer as local_size / 2 (assuming local_size is a power of 2).